docs/design/keyboard_handler.md

```diff
@@ -0,0 +1,87 @@
+# Keyboard handler
+Package providing ability to handle keyboard input via simple interface with callbacks. 
+
+## Goal
+We need to be able to handle keyboard input in unified way with cross-platform implementation.
+
+## Design Proposal
+It would be great to have some simple interface to be able to subscribe to the specific key 
```

Comment:
This is good - for readability I would put this as a list of **design requirements**:

```
Design requirements:
* Subscribe to keyboard events via callbacks
* Subscriptions work with key modifiers (e.g. Ctrl+C)
* Multiple clients can subscribe to the same event
```